src/save.ts

```diff
@@ -34,6 +34,15 @@ async function run(): Promise<void> {
             return;
         }
 
+        core.debug("Reserving Cache");
+        const cacheId = await cacheHttpClient.reserveCache(primaryKey);
+        if (cacheId < 0) {
```

Comment:
Double-checking, zero is valid here?

Reply:
The cacheId starts incrementing from 1, so the server would never return 0.

Reply:
Changed this to explicitly test `if (cacheId == -1)` which is the default return value in the method when no cache ID was sent back.